bin/common.c

```diff
@@ -75,3 +77,114 @@ int key_log_callback(void *file, struct s2n_connection *conn, uint8_t *logline,
 
     return fflush((FILE *)file);
 }
+
+/* An inverse map from an ascii value to a hexadecimal nibble value
+ * accounts for all possible char values, where 255 is invalid value */
+static const uint8_t hex_inverse[256] = {
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+      0,   1,   2,   3,   4,   5,   6,   7,   8,   9, 255, 255, 255, 255, 255, 255,
+    255,  10,  11,  12,  13,  14,  15, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255,  10,  11,  12,  13,  14,  15, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255
+};
+
+int s2n_str_hex_to_bytes(const uint8_t *hex, uint8_t *out_bytes, uint32_t *out_bytes_len)
+{
+    GUARD_EXIT_NULL(hex);
+    GUARD_EXIT_NULL(out_bytes);
+    GUARD_EXIT_NULL(out_bytes_len);
+
+    uint32_t len_with_spaces = strlen((const char*)hex);
+    size_t i = 0, j = 0;
+    while (j < len_with_spaces) {
+        if (hex[j] == ' ') {
+            j++;
+            continue;
+        }
+
+        uint8_t high_nibble = hex_inverse[hex[j]];
+        if (high_nibble == 255) {
+            fprintf(stderr, "Invalid HEX encountered\n");
+            return S2N_FAILURE;
+        }
+
+        uint8_t low_nibble = hex_inverse[hex[j + 1]];
+        if (low_nibble == 255) {
+            fprintf(stderr, "Invalid HEX encountered\n");
+            return S2N_FAILURE;
+        }
+
+        if(*out_bytes_len < i) {
+            fprintf(stderr, "Insufficient memory for bytes buffer, try increasing the allocation size\n");
+            return S2N_FAILURE;
+        }
+        out_bytes[i] = high_nibble << 4 | low_nibble;
+
+        i++;
+        j+=2;
+    }
+    *out_bytes_len = i;
+
+    return S2N_SUCCESS;
+}
+
+static int s2n_get_psk_hmac_alg(s2n_psk_hmac *psk_hmac, char *hmac_str)
+{
+    GUARD_EXIT_NULL(psk_hmac);
+    GUARD_EXIT_NULL(hmac_str);
+
+    if (strcmp(hmac_str, "S2N_PSK_HMAC_SHA256") == 0) {
+        *psk_hmac = S2N_PSK_HMAC_SHA256;
```

Comment:
Why is the string you expect prefixed with "S2N_PSK_HMAC"? Why not just "SHA256"?